### ISSUE TITLE
How can I accept bitcoin using NBitcoin?

### ISSUE THREAD

**Issue by ScofieldYeh** [MetacoSA/NBitcoin#386]

**Comment by ScofieldYeh** [comment]
Thank you so much for your fast reply and your great work to let me understand the bitcoin programming in a few days. 
I'd like build service like GoURL.io for my website.
So I need to create  P2PKH address for my clients to pay then I use NBXp to track the UTXOs status, Correct?
Is there any method I can leverage to instant confirm the transaction to prevent my client wait 15 mins or more? Or it's still not possible in current environment?
Another question is do I need to rebuild your ProgrammingBitcoinFunding website to learn more about the bticoin coding? I found the datebase ProgrammingBitcoinFunding-20150114124016.mdf is not included in your files.

Thanks again for your kind help.

**Comment by NicolasDorier** [comment]
No that's fine you don't need this.

Yes you use NBx to track UTXOs. You can't instant confirm, the merchant should decide to take the risk to be double spent if he wishes.

If you just want to help a merchant, just use BTCPay which is a service self hostable for merchant (it uses nbx under the hood):

* https://www.youtube.com/channel/UCpG9WL6TJuoNfFVkaDMp9ug
* https://github.com/btcpayserver/btcpayserver

**Comment by nopara73** [comment]
@ScofieldYeh 

If I'd be you, I'd just use a public API: https://www.smartbit.com.au/api  
Alongside with my KeyManager: https://github.com/nopara73/MagicalCryptoWallet/blob/master/MagicalCryptoWallet/KeyManagement/KeyManager.cs

**Comment by NicolasDorier** [comment]
Public API can't query over a xpub. You need to query all addresses of your wallets to detect payments.

**Comment by nopara73** [comment]
Many of them can, even the one you wrote (QBitNinja.) That being said, it's not like it'd be needed in this case, he can periodically query only one address, that's just about to get paid. Or even better, SmartBit has websocket notifications for addresses.

1. Generate new address.  
2. Subscribe to websocket notification for the address.  
3. Wait until the customer pays.  
4. Unsubscribe.

**Comment by NicolasDorier** [comment]
NBxplorer has also websocket notification. (https://github.com/dgarage/NBXplorer/blob/master/docs/API.md)

**Comment by nopara73** [comment]
But you must run a full node with that, don't you?

**Comment by NicolasDorier** [comment]
yes indeed. So this depends on whether this is on the table or not.

**Comment by ScofieldYeh** [comment]
I am curious about why we use address to monitor or query the transaction results instead of the transaction IDs? Especially I would like to know how many blocks of a transaction is confirmed?

**Comment by NicolasDorier** [comment]
how can the payment server know about the transaction id?

**Comment by ScofieldYeh** [comment]
BroadcastResponse broadcastResponse = client.Broadcast(transaction).Result;

            if (!broadcastResponse.Success)
            {
                Console.Error.WriteLine("ErrorCode: " + broadcastResponse.Error.ErrorCode);
                Console.Error.WriteLine("Error message: " + broadcastResponse.Error.Reason);
            }
            else
            {
                Console.WriteLine("Success! You can check out the hash of the transaciton in any block explorer:");
                Console.WriteLine(transaction.GetHash());
            }

from your code can we use transaction.GetHash() to query how many confirmed it get?

**Comment by NicolasDorier** [comment]
yes
